### PR TITLE
Make SlingShot skill

### PR DESCRIPTION
**Pull request by mrlukasbos**

### CODE REVIEW

File: roboteam_ai/src/skills/SlingShot.cpp [hunk 1]
```diff
@@ -0,0 +1,126 @@
...
+    }
+    progression = DRIBBLING;
+    rotateAngle=robot->angle;
+    kickOrient=robot->angle;
+}
+Skill::Status SlingShot::onUpdate() {
+    progression=updateProgress(progression);
+
+    switch(progression){
+    case DRIBBLING: std::cout<<"dribbling"<<std::endl; break;
```
Comment (ThijsLuttikhuis) [inline roboteam_ai/src/skills/SlingShot.cpp]: remove prints

File: roboteam_ai/src/skills/SlingShot.cpp [hunk 2]
```diff
@@ -0,0 +1,126 @@
...
+    return Status::Failure;
+}
+void SlingShot::onTerminate(Skill::Status status) {
+
+}
+SlingShot::Progression SlingShot::updateProgress(Progression currentProgress) {
+    int maxDribbleTicks = 200;
+    int maxWaitingTicks= 300;
+    // we go from dribbling to rotating away to waiting for a result.
+    if (currentProgress == DRIBBLING) {
```
Comment (mrlukasbos) [inline roboteam_ai/src/skills/SlingShot.cpp]: This function is messy.

File: roboteam_ai/src/skills/SlingShot.cpp [hunk 3]
```diff
@@ -0,0 +1,126 @@
...
+    case FAIL: sendWaitCommand(); return Status::Failure;
+    case SUCCESS: sendWaitCommand();return Status::Failure;
+    }
+    return Status::Failure;
+}
+void SlingShot::onTerminate(Skill::Status status) {
+
+}
+SlingShot::Progression SlingShot::updateProgress(Progression currentProgress) {
+    int maxDribbleTicks = 200;
```
Comment (mrlukasbos) [inline roboteam_ai/src/skills/SlingShot.cpp]: make them const int and all caps (and preferebly in header file)

File: roboteam_ai/src/skills/SlingShot.cpp [hunk 4]
```diff
@@ -0,0 +1,126 @@
...
+    switch (progression){
+    case DRIBBLING: sendDribbleCommand(); return Status::Running;
+    case ROTATINGAWAY: sendRotateCommand(); return Status::Running;
+    case WAITINGFORRESULT: sendWaitCommand();return Status::Running;
+    case FAIL: sendWaitCommand(); return Status::Failure;
+    case SUCCESS: sendWaitCommand();return Status::Failure;
+    }
+    return Status::Failure;
+}
+void SlingShot::onTerminate(Skill::Status status) {
```
Comment (mrlukasbos) [inline roboteam_ai/src/skills/SlingShot.cpp]: remove the onTerminate

File: roboteam_ai/src/skills/SlingShot.cpp [hunk 5]
```diff
@@ -0,0 +1,126 @@
...
+
+    switch(progression){
+    case DRIBBLING: std::cout<<"dribbling"<<std::endl; break;
+    case ROTATINGAWAY: std::cout<<"RotatingAway"<<std::endl; break;
+    case WAITINGFORRESULT: std::cout<<"Waiting"<<std::endl; break;
+    case FAIL: std::cout<<"FAIL"<<std::endl; break;
+    case SUCCESS: std::cout<<"SUCCESS"<<std::endl; break;
+
+    }
+    switch (progression){
```
Comment (mrlukasbos) [inline roboteam_ai/src/skills/SlingShot.cpp]: I don't like doing two things on one line. Refactor with brackets plz

File: roboteam_ai/src/skills/SlingShot.cpp [hunk 6]
```diff
@@ -0,0 +1,123 @@
...
+            dribbledTicks ++;
+            return world::world->robotHasBall(robot->id, true) ? DRIBBLING : FAIL;
+        }
+        else {
+            setRotate();
+            return ROTATINGAWAY;
+        }
+    }// if the robot is at the angle we start waiting
+    else if (currentProgress == ROTATINGAWAY) {
+        return robotAtAngle() ? WAITINGFORRESULT : ROTATINGAWAY;;
```
Comment (ThijsLuttikhuis) [inline roboteam_ai/src/skills/SlingShot.cpp]: ;; ;;;;; ;;; ;    ;;;
;;;;; ; ;;

**Pull request merged**